doc/source/maintenance/process.rst

```diff
@@ -0,0 +1,90 @@
+Maintenance Process
+===================
+
+This section provide guidances and process to eventlet
+maintainers. They are useful to lead the life cycle of eventlet.
+
+Releases
+--------
+
+Here we describe the process we usually follow to
+process a new release.
+
+1. Create a github issue to track the release
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The first step will be to `open a new github issue`_
+to warn other maintainers about our intention
+to produce a new release. They may want, or not,
+to land a specific patch to address a specific
+topic. This issue will allow them to raise their
+concerns.
+
+Here are some `previous examples of issues`_ specifically
+created to handle the release process. Usually we name this
+kind of issue with the following pattern "[release] eventlet <next-version-number>".
+
+Please add the `release` label to this issue. It would
+ease the tracking of works related to releases.
+
+2. Prepare the changelog
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+You now have to update the changelog by updating
+the `NEWS` file available at the root of eventlet the project.
+
+We would recommand to give the big picture of the changes
+landed by the coming version. The goal here is not to list
+each commit, but rather, to give a summarize of the significant
+changes made during this versions.
+
+Once your changes are done, then propose a pull request.
+
+Please add the `changelog` label to this pull request. It would
+ease the tracking of works related to releases.
+
+If you want, you can use the issue previously created to list
+each commits landed in this new version. Here is an example https://github.com/eventlet/eventlet/issues/897.
+
+3. Create the tag
+~~~~~~~~~~~~~~~~~
+
+Once the changelog patch is merged, then we are now
+able to produce the new corresponding tag, here are the
+commands we use to do that:
+
+```bash
```

Comment:
It's also easy to do this from the GitHub UI.

Reply:
Indeed, I added a sentence related to github UI